### DISCUSSION TITLE
Class and style attributes on all components

### DISCUSSION THREAD

**Discussion by Eragra3** [thaw-ui/thaw#219] [Q&A]
Hello, I'm new to Leptos and your library, but does it make sense to expect all components to support basic html attributes like `class`, `style` maybe `id`?

Assuming they have some html container element in `view! {}`. No specific logic, just passthrough to component top-most element. That would simplify styling, especially for things like Tailwind.

**Comment by lizidev** [comment]
> but does it make sense to expect all components to support basic html attributes like class, style maybe id?

I think this is a very reasonable requirement, especially for Vue developers.

> Assuming they have some html container element in view! {}. No specific logic, just passthrough to component top-most element. That would simplify styling, especially for things like Tailwind.

This feature will be fully implemented in Thaw-v0.4. Then you can use the following methods:

```rust
/// class
<div class=""> or <div attr:class="">
/// style
<div style=""> or <div attr:style="">
/// id
<div attr:id="">
```
